squidDrone/src/i2c/i2c.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef SRC_I2C_HANDLER_HPP_
+#define SRC_I2C_HANDLER_HPP_
+
+#include "stm32g4xx_hal.h"
+#include "i2c_config.h"
+#include "i2c_interface.hpp"
+
+namespace i2c {
+
+  /**
+   * @brief The concrete implementation of I2C Interface
+   * 
+   */
+  class I2C final : public I2CInterface {
+    public:
+      ~I2C() = default;
+      explicit I2C(void): I2CInterface(){};
+
+      auto Read(uint8_t address, uint16_t byte_size, uint32_t timeout=I2C_STANDARD_TIMEOUT_IN_MS) noexcept -> std::tuple<I2CStatus, std::vector<uint8_t>> override;
+      auto Write(uint8_t address, const std::vector<uint8_t>& data, uint32_t timeout=I2C_STANDARD_TIMEOUT_IN_MS) noexcept -> I2CStatus override;
+    private:
+      auto CheckForValidInputRead(uint8_t address, uint16_t byte_size, uint32_t timeout) noexcept -> bool;
```

Comment:
Technically you don't have to document your private functions, but it is immensely helpful for understanding the code.
It is up to you.

Reply:
I thought also about this, but from "Clean Code" from Robert C. Martin, it would be white noise, if the name of the method is clear.
E.g.: CheckIfI2CAddressIsValid /// checks if i2c address is valid

Reply:
Ok in this case I agree with you, because the methods are simple checker methods.